### DISCUSSION TITLE
Improvements in issue reporting

### DISCUSSION THREAD

**Discussion by TylerLeonhardt** [microsoft/vscode-discussions#119] [Announcements]
Hi all 👋 

I just wanted to let you know about a couple of improvements we've made in issue reporting to help with the quality of issues that are opened.

The goal, was to minimize issues like this... which aren't very helpful, we know:
<img width="917" alt="image" src="https://user-images.githubusercontent.com/2644648/187814237-af207d05-3cdc-4020-9acc-90cd020dd68c.png">

### Use the issue reporter everywhere

First off, @DanTup helped us realize that these issues weren't created by our built-in Issue Reporter. They were created here:

![image](https://user-images.githubusercontent.com/2644648/187815017-839c28c8-f09a-4033-a3f2-6c8e238b7a65.png)
which you can get to via `>Developer: Show Running Extensions` and it used some really old code that likely pre-dated the Issue Reporter...

So, with https://github.com/microsoft/vscode/pull/159707 we no longer use this old logic and instead route users to the Issue Reporter. This should dramatically decrease the amount of "please paste" issues you see open on your repos.

### Allow for larger "steps to reproduce" in the Issue Reporter

After discussions with GitHub, we've realized that our 2500 character limit on the URL opened by the issue reporter could be increased. We've increased this limit to 7500 in https://github.com/microsoft/vscode/pull/159491. This should also decrease the number of "please paste" issues that you see.

There's still a lot more that can be done for issue reporting, but we hope that these little improvements will enable more quality issues for all of us 😃

**Comment by Spyder-0** [comment]
Very cool 👍

**Comment by bwateratmsft** [comment]
Related to the issue reporter, this is another improvement we'd love to see @TylerLeonhardt: https://github.com/microsoft/vscode/issues/46726 and/or https://github.com/microsoft/vscode/issues/148675

As an example, in the Docker extension we use an extension-supplied issue reporter, so that we can supply the call stack, error name, etc. in some reported issues. We'd really prefer to use the built-in issue reporter, though. However, we also see a lot of issues reported with no helpful user-provided information like repro steps; usually these issues are unactionable. Ideally, what we'd like is to include the stack and error info but _still_ force users to type _something_ into the Repro Steps field. If we put this extra info into the repro steps field, they are not required to enter anything--and usually don't.

**Comment by christian-bromann** [comment]
I never knew VS Code had an internal UI for reporting issues. I would love to see this more elevated, e.g. when doing a right click on the "Installed Extension" in the sidebar:

![Screenshot 2022-09-06 at 18 02 00](https://user-images.githubusercontent.com/731337/188682648-ef1a9575-e4a5-4e07-9cce-499a1ec2e762.png)

And maybe even on the extension main page:

![Screenshot 2022-09-06 at 18 03 09](https://user-images.githubusercontent.com/731337/188682896-665c80d5-451b-4ea9-bafb-8711f1415b69.png)

**Reply by TylerLeonhardt** [reply]
This sounds good with https://github.com/microsoft/vscode/issues/159786
